x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -250,26 +256,26 @@ function parseAndVerifyReadme(paths: string[], pkgName: string, pkgVersion: stri
 export function parseAndVerifyDataStreams(
   paths: string[],
   pkgName: string,
-  pkgVersion: string
+  pkgVersion: string,
+  pkgBasePathOverride?: string
```

Comment:
allow us to override the base path of the package to validate, all this code was expecting a certain directory strucutre which we see in prod. 

Also did some small refactoring of this code while I was here :)